kibana.md

```diff
@@ -1101,3 +1101,201 @@ an admin, who has just configured access for a new user. They would like
+    mutations.forEach(mutation => {
+      mutation.addedNodes.forEach(node => {
+        const customLogo = document.querySelector('#customLogo');
+
+        const createCustomLogo = () => {
+          const img = document.createElement('img');
+          img.src = `data:image/svg+xml;base64,${window.ROR_METADATA.newLogo}`;
+          img.style.width = '32px';
+          img.style.height = '32px';
+          img.id = 'customLogo';
+          logoHeader.appendChild(img);
+        };
+
+        const hideAllLogoElements = () => {
+          Array.from(logoHeader.childNodes).forEach(node => {
+            node.style.display = 'none';
+          });
+        };
+
+        const handleInit = () => {
+          hideAllLogoElements();
+          createCustomLogo();
+        };
+
+        if (customLogo) {
+          const displayCustomLogo = () => {
+            customLogo.style.display = 'block';
+          };
+          const hideCustomLogo = () => {
+            customLogo.style.display = 'none';
+          };
+          if (node.role === 'progressbar') {
+            hideCustomLogo();
+          }
+
+          if (node.role === 'img') {
+            const hideDefaultLogo = () => {
+              node.style.display = 'none';
+            };
+
+            hideDefaultLogo();
+            displayCustomLogo();
+          }
+        }
+
+        if (node.dataset.type === 'logoElastic' && !customLogo) {
+          handleInit();
+        }
+      });
+    });
+  });
+
+  observer.observe(logoHeader, { childList: true });
+}
+```
+All session metadata will be available via `window.ROR_METADATA` property. To get your custom logo, just use `window.ROR_METADATA.newLogo` value. In the example above, after login in as a user with username `admin` you will see a custom logo.
+The whole example is a little complex but seems, kibana logo is also a loading indicator, we need to detect the loading state, replace the logo with a spinner, and after the loading, back the custom logo again.
+
+#### Reject machine to machine traffic using custom metadata ACL rules
+We can also reject the specific request for example based on the custom metadata
+
+1. Define ACL in your `readonlyrest.yml` file
+
+```yaml
+  - name: ADMIN_GRP
+    groups: [ administrators ]
+    kibana:
+       access: admin
+       index: '.kibana_@{acl:current_group}'
+       metadata:
+          rejectBasicAuth: true
+```
+
+2. Declare custom kibana js file `readonlyrest_kbn.kibana_custom_js_inject_file: '/path/to/custom_kibana.js'`
+
+```js
+async function customMiddleware(req, res, next) {
+   const metadata =
+           req.rorRequest && req.rorRequest.getIdentitySession() && req.rorRequest.getIdentitySession().metadata;
+
+   const headerAuth = req.rorRequest && req.rorRequest.getAuthorizationHeaders && req.rorRequest.getHeaders().getAuthorizationHeaders().get('authorization');
+   const basicHeaderAuth = headerAuth && headerAuth.includes('Basic')
```

Comment:
this var should be called `isBasicAuth` because it's boolean

Reply:
Done ✔️